File: app/src/main/java/com/nvvi9/giphyme/domain/mappers/FavoriteGifItemMapper.java
package com.nvvi9.giphyme.domain.mappers;

import com.nvvi9.giphyme.data.BaseMapper;
import com.nvvi9.giphyme.data.gifinfo.GifInfo;
import com.nvvi9.giphyme.vo.FavoriteGifItem;

public class FavoriteGifItemMapper implements BaseMapper<GifInfo, FavoriteGifItem> {

    @Override
    public FavoriteGifItem map(GifInfo value) {
        return new FavoriteGifItem(value.getId(), value.getTitle(), value.getGifUrl(), value.getGifUrl(), value.getAuthorName(), value.getAuthorAvatarUrl());
    }
}

File: app/src/main/java/com/nvvi9/giphyme/domain/mappers/FavoriteGifItemMapper.kt
package com.nvvi9.giphyme.domain.mappers

import com.nvvi9.giphyme.data.BaseMapper
import com.nvvi9.giphyme.data.gifinfo.GifInfo
import com.nvvi9.giphyme.vo.FavoriteGifItem

class FavoriteGifItemMapper : BaseMapper<GifInfo, FavoriteGifItem> {

    override fun map(value: GifInfo): FavoriteGifItem =
        FavoriteGifItem(
            value.id,
            value.title,
            value.gifUrl,
            value.gifUrl,
            value.authorName,
            value.authorAvatarUrl
        )
}